src/ophys_etl/decrosstalk/decrosstalk_schema.py

```diff
@@ -0,0 +1,121 @@
+                        argschema.fields.List(argschema.fields.Bool),
+                        description='mask',
+                        cli_as_single_argument=True)
+
+
+class PlaneSchema(argschema.ArgSchema):
+
+    ophys_experiment_id = argschema.fields.Int(
+                             description='experiment ID',
+                             required=True)
+
+    motion_corrected_stack = argschema.fields.Str(
+                        description='path to motion corrected movie',
+                        required=True)
+
+    motion_border = argschema.fields.Dict(description='motion border',
+                                          required=True)
+
+    rois = argschema.fields.List(
+                argschema.fields.Nested(RoiSchema),
+                description='regions of interest',
+                required=True,
+                cli_as_single_argument=True)
+
+
+class PlanePairSchema(argschema.ArgSchema):
+
+    ophys_imaging_plane_group_id = argschema.fields.Int(
+                                    description='group ID',
+                                    required=True)
+
+    group_order = argschema.fields.Int(
+                      description='order of group in session',
+                      required=True)
+
+    planes = argschema.fields.List(
+                 argschema.fields.Nested(PlaneSchema),
+                 description='planes',
+                 required=True,
+                 cli_as_single_argument=True)
+
+
+class DecrosstalkInputSchema(argschema.ArgSchema):
+
+    ophys_session_id = argschema.fields.Int(
+                           description='ophys_session_id',
+                           required=True)
+
+    qc_output_dir = argschema.fields.Str(
+                        description='path to intermediate output dir',
+                        required=True)
+
+    coupled_planes = argschema.fields.List(
+                        argschema.fields.Nested(PlanePairSchema),
+                        description='list of plane pairs',
+                        required=True,
+                        cli_as_single_argument=True)
+
+
+def validate_list_of_ints(input_list):
+    """
+    Validator for DecrosstalkOutputSchema.
+    Checks that input_list is, indeed, a list of ints.
+    """
+    if not isinstance(input_list, list):
+        return False
+    for ii in input_list:
+        if not isinstance(ii, int):
+            msg = 'Did not contain ints'
+            raise marshmallow.exceptions.ValidationError(msg)
+    return True
+
+
+class DecrosstalkOutputSchema(argschema.ArgSchema):
+    # All Lists are lists of argschema.fields.Field because
+    # this was the only way I could get argschema to not
+    # automatically cast non-int numeric types into ints.
+    # The call to validate_list_of_ints will make sure that
+    # the lists do, indeed, need to contain ints.
+
+    decrosstalk_raw_exclusion_label = argschema.fields.List(
+                                       argschema.fields.Field,
```

Comment:
another way could be to use a marshmallow pre_load decorated function to check inputs before parsing.

Reply:
I'm still very unfamiliar with how marshmallow and argschema work. Given that this function meets the need, do you mind if I leave it as-is? We can revisit this in a future ticket if necessary.

Reply:
it works, leave it